src/openforms/submissions/tests/factories.py

```diff
@@ -126,6 +139,10 @@ class Params:
                 status=PaymentStatus.registered,
             ),
         )
+        with_public_registration_reference = factory.Trait(
+            registration_success=True,
+            public_registration_reference=factory.LazyFunction(get_random_reference),
+        )
```

Comment:
I think this implies `completed=True` too since a public registration reference can only created from the on-post-completion event.